synapse/storage/registration.py

```diff
@@ -598,11 +601,71 @@ def __init__(self, db_conn, hs):
             "user_threepids_grandfather", self._bg_user_threepids_grandfather,
         )
 
+        self.register_background_update_handler(
+            "users_set_deactivated_flag", self._set_deactivated_flag,
+        )
+
         # Create a background job for culling expired 3PID validity tokens
         hs.get_clock().looping_call(
             self.cull_expired_threepid_validation_tokens, THIRTY_MINUTES_IN_MS,
         )
 
+    @defer.inlineCallbacks
+    def _set_deactivated_flag(self, progress, batch_size):
```

Comment:
Sorry, sorry, really sorry, but can you call this `_backgroud_update_set_deactivated_flag` or something and add a really quick docstring along the lines of "Goes and fills out the user deactivated status flag for existing deactivated users"